src/styles/blocks/calendar.scss

```diff
@@ -0,0 +1,51 @@
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  width: $calendar_size;
+  gap: $gap;
+  padding: 20px;
+
+  @for $index from 28 through 31 {
+    &--month-length--#{$index} {
+      .calendar__days:nth-child(n + #{$index+1}) {
+        display: none;
+      }
+    }
+  }
+
+  @each $day, $count in $days {
+    &--start-day--#{$day} {
+      .calendar__days:first-child {
+        margin-left: $cell_size * $count + $gap * $count;
+      }
+    }
+  }
+
+  @for $index from $days_per_month through 1 {
+
+    &__days--day--#{$index}::before {
+      content: "#{$index}";
+    }
+  }
+
+  &__days {
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    width: $cell_size;
+    height: $cell_size;
+    padding: $horizontal_padding;
+    text-align: center;
+    background-color: $base_color;
+    border: 1px solid black;
+    transition: transform 0.5s ease-in-out;
+    cursor: pointer;
+  }
+
+  @for $index from 1 through $days_per_month {
+    &__days--day--#{$index}:hover {
+      background-color: $hover_color;
+      transform: translateY(-20px);
+    }
+  }
```

Comment:
why do you add 30 different `hover` classes with your loop? just add this hover to your `days` class